### ISSUE TITLE
aws_ecs_cluster: Networking configuration missing (VPC+Subnets)

### ISSUE THREAD

**Issue by matei-dragu** [hashicorp/terraform-provider-aws#24288]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

When creating a `aws_ecs_cluster` resource you cannot specify the VPC/Subnets to use.

This behavious is happening in version `4.10.0` of the AWS provider. 

### New or Affected Resource(s)

* aws_ecs_cluster

### Potential Terraform Configuration

```hcl
resource "aws_ecs_cluster" "one_of_the_clusters_in_my_aws_account" {
  name = "one_of_the_clusters_in_my_aws_account"
  
  # here we should have the option to explicitly specify:
  # 1. The VPC inside which the cluster will be created
  # 2. The subnet(s) inside that VPC that the cluster will use
}
```

### References

Basically the config options in the `Networking` section are missing:

![Screenshot 2022-04-17 at 12 50 49](https://user-images.githubusercontent.com/101947107/163713259-2555ff16-d7a7-4153-8139-a126df59a47e.png)

**Comment by trevorrea** [comment]
Hi @ceresdarkmatter 

I don't use ECS + EC2 but the Networking settings only apply for that and not Fargate. With Fargate this option is not available.

I guess with ECS + EC2 the wizard asks for those options so it can create an autoscaling group with the right settings whereas in Terraform these are all separate resources and you have to create all the supporting resources yourself. 

In Terraform this would be done like this:-

aws_ecs_cluster - https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ecs_cluster
aws_ecs_capacity_provider - https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ecs_capacity_provider
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/autoscaling_group - it is here were you would specify the network settings not on the ECS cluster resource.

There may be some more resources required, as I said above I don't use ECS + EC2 as I use Fargate (it's really cool :) ) but hope this helps.

**Comment by epomatti** [comment]
Perhaps @trevorrea explanation could be added to the documentation to avoid confusion?

**Comment by trevorrea** [comment]
Feel free to open an MR :) though it's all in the AWS docs.

To make it slightly clearer an autoscaling group could be added to the example at [https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ecs_cluster#example-with-capacity-providers]()
